src/Linter.php

```diff
@@ -89,4 +99,43 @@ private function packagesAreSorted(array $packages)
 
         return $sortedNames === $names;
     }
+
+    /**
+     * @param string[] $packages
+     *
+     * @return array
+     */
+    private function validateVersionConstraints(array $packages)
+    {
+        $errors = array();
+
+        foreach ($packages as $name => $constraint) {
+            $errorPrefix = sprintf("Requirement '%s:%s' ", $name, $constraint);
+
+            // Checks if OR format is correct
+            // From Composer\Semver\VersionParser::parseConstraints
+            $orConstraints = preg_split('{\s*\|\|?\s*}', trim($constraint));
+            foreach ($orConstraints as &$subConstraint) {
+                // Checks ~ usage
+                $subConstraint = str_replace('~', '^', $subConstraint);
+
+                // Checks for usage like ^2.1,>=2.1.5. Should be ^2.1.5.
+                // From Composer\Semver\VersionParser::parseConstraints
+                $andConstraints = preg_split('{(?<!^|as|[=>< ,]) *(?<!-)[, ](?!-) *(?!,|as|$)}', $subConstraint);
+                if (2 === count($andConstraints) && '>=' === substr($andConstraints[1], 0, 2)) {
+                    $andConstraints[1] = '^'.substr($andConstraints[1], 2);
+                    array_shift($andConstraints);
+                    $subConstraint = implode(',', $andConstraints);
+                }
+            }
+
+            $expectedConstraint = implode(' || ', $orConstraints);
+
+            if ($expectedConstraint !== $constraint) {
+                array_push($errors, $errorPrefix."format is not valid. Should be '".$expectedConstraint."'.");
```

Comment:
Instead of putting the space at the end of `$errorPrefix` you could but it in this string and simply write the very natural : `"$errorPrefix format is not valid. Should be '$expectedConstraint'."` I've found myself avoiding this kind of thing for years but now I try to always think a pick the most simple solution between string interpolation, concatenation and sprintf. What do you think?